sorbet/rbi/shims/rubygems.rbi

```diff
@@ -0,0 +1,34 @@
+# typed: strong
+# frozen_string_literal: true
+
+module Gem
+  class Version
```

Comment:
optional: Might be worth adding a comment about the fact that we're overriding/narrowing the underlying signature (essentially what you have in the PR description)?